### PR TITLE
[RHELC-1334] Port ModifiedRPMFilesDiff to Action framework

### PR DESCRIPTION
**Pull request by hosekadam**
This new action provides information about difference between 'rpm -Va' from before and after the conversion.

Durign this port was needed to change the logging in the SystemInfo class to use global provided logging instead of the one inside of the class. Which was overall the same, just created dependency for logging to be inside of the class, or in the systeminfo.py would be duplicity for logging.

Jira Issues:

-[RHELC-1334](https://issues.redhat.com/browse/RHELC-1334)

Checklist

- [x] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [x] `[RHELC-]` or `[HMS-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: convert2rhel/systeminfo.py [hunk 1]
```diff
@@ -549,5 +500,26 @@ def is_systemd_managed_service_running(service):
...
+    logger.info(
+        "Running the 'rpm -Va' command which can take several"
+        " minutes. It can be disabled by using the"
+        " --no-rpm-va option."
+    )
+    rpm_va, _ = utils.run_subprocess(["rpm", "-Va"], print_output=False)
+    output_file = os.path.join(LOG_DIR, log_filename)
+    utils.store_content_to_file(output_file, rpm_va)
+    logger.info("The 'rpm -Va' output has been stored in the %s file." % output_file)
+
```
Comment (r0x0d) [inline convert2rhel/systeminfo.py:522]: To minimize the changes to the systeminfo that doesn't seem related to the code, why don't we keep this inside the class and only call system_info.generate_rpm_va() inside the action classs? That would minimize all of those logger changes

Comment (r0x0d) [reply convert2rhel/systeminfo.py:522]: To expand a bit on this, I think this is a good change, moving the logger instance to be outside the class is definitely a way to go. But it also feels unrelated to the scope of the ticket. We could open a new PR that patches this as well.

File: convert2rhel/actions/post_conversion/modified_rpm_files_diff.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            logger.info("Skipping comparison of the 'rpm -Va' output from before and after the conversion.")
+            self.set_result(
+                level="SKIP",
+                id="SKIPPED_MODIFIED_RPM_FILES_DIFF",
+                title="Skipped comparison of 'rmp -Va' output from before and after the conversion.",
+                description="Comparison of 'rpm -Va' output was skipped due missing output "
+                "of the 'rpm -Va' run before the conversion.",
+                diagnosis="This is caused mainly by using '--no-rpm-va' argument for convert2rhel.",
+            )
+            return
```
Comment (r0x0d) [inline convert2rhel/actions/post_conversion/modified_rpm_files_diff.py:56]: I'm not sure if the use of the `skip` level here is the following the same pattern. We usually set `SKIP` automatically if any action dependant on this one fails. I think I will to double check.

**Review by r0x0d — COMMENTED** [review]
Looks good. Some minors coments here and there. Also, will need to double check the usage of level = `SKIP`.

File: convert2rhel/systeminfo.py (hunk 1 above)
Comment (hosekadam) [reply convert2rhel/systeminfo.py:522]: Okay. I wanted to take the `generate_rpm_va` outside the class. But you are right, this is unrelated to the ticket. I'll open a new PR with the changes.

File: convert2rhel/actions/post_conversion/modified_rpm_files_diff.py (hunk 2 above)
Comment (hosekadam) [reply convert2rhel/actions/post_conversion/modified_rpm_files_diff.py:56]: Soo, it should be `INFO`?

Comment (hosekadam) [reply convert2rhel/actions/post_conversion/modified_rpm_files_diff.py:56]: I see the info cannot be the result. From the possible values (success, skip, fail) I like the most skip. But not sure about the ussage you are mentioning

**Comment by hosekadam** [comment]
Applied the changes - no big changes to the systeminfo, porting (better say writing new) unit tests. Still WIP, will continue after PTO

File: convert2rhel/actions/post_conversion/modified_rpm_files_diff.py (hunk 2 above)
Comment (r0x0d) [reply convert2rhel/actions/post_conversion/modified_rpm_files_diff.py:56]: I commented in our slack channel about that. For now, it is totally fine to have this as skip

File: convert2rhel/actions/post_conversion/modified_rpm_files_diff.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+        system_info.generate_rpm_va(log_filename=POST_RPM_VA_LOG_FILENAME)
+
+        pre_rpm_va_log_path = os.path.join(LOG_DIR, PRE_RPM_VA_LOG_FILENAME)
+        if not os.path.exists(pre_rpm_va_log_path):
+            logger.info("Skipping comparison of the 'rpm -Va' output from before and after the conversion.")
+            self.set_result(
+                level="SKIP",
+                id="SKIPPED_MODIFIED_RPM_FILES_DIFF",
+                title="Skipped comparison of 'rmp -Va' output from before and after the conversion.",
```
Comment (r0x0d) [inline convert2rhel/actions/post_conversion/modified_rpm_files_diff.py:51]:
```suggestion
                title="Skipped comparison of 'rpm -Va' output from before and after the conversion.",
```

File: convert2rhel/actions/post_conversion/modified_rpm_files_diff.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            logger.info("Skipping comparison of the 'rpm -Va' output from before and after the conversion.")
+            self.add_message(
+                level="INFO",
+                id="SKIPPED_MODIFIED_RPM_FILES_DIFF",
+                title="Skipped comparison of 'rpm -Va' output from before and after the conversion.",
+                description="Comparison of 'rpm -Va' output was skipped due missing output "
+                "of the 'rpm -Va' run before the conversion.",
+                diagnosis="This is caused mainly by using '--no-rpm-va' argument for convert2rhel.",
+            )
+            return
```
Comment (r0x0d) [inline convert2rhel/actions/post_conversion/modified_rpm_files_diff.py:56]: Just to avoid having different messages in the waring and the message.

```suggestion
            title = "Skipping comparison of the 'rpm -Va' output from before and after the conversion."
            logger.info(title)
            self.add_message(
                level="INFO",
                id="SKIPPED_MODIFIED_RPM_FILES_DIFF",
                title=title,
                description="Comparison of 'rpm -Va' output was skipped due missing output "
                "of the 'rpm -Va' run before the conversion.",
                diagnosis="This is caused mainly by using '--no-rpm-va' argument for convert2rhel.",
            )
            return
```

File: convert2rhel/actions/post_conversion/modified_rpm_files_diff.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            logger.info(
+                "Comparison of modified rpm files from before and after the conversion:\n%s" % modified_rpm_files_diff
+            )
+            self.add_message(
+                level="INFO",
+                id="FOUND_MODIFIED_RPM_FILES",
+                title="Modified rpm files from before and after the conversion were found.",
+                description="Comparison of modified rpm files from before and after "
+                "the conversion: \n%s" % modified_rpm_files_diff,
+            )
```
Comment (r0x0d) [inline convert2rhel/actions/post_conversion/modified_rpm_files_diff.py:82]:
```suggestion
            description = "Comparison of modified rpm files from before and after the conversion:\n%s" % modified_rpm_files_diff
            logger.info(description)
            self.add_message(
                level="INFO",
                id="FOUND_MODIFIED_RPM_FILES",
                title="Modified rpm files from before and after the conversion were found.",
                description=description,
            )
```

**Review by r0x0d — APPROVED** [review]
Looks good to me

**Review by r0x0d — APPROVED** [review]

File: convert2rhel/actions/post_conversion/modified_rpm_files_diff.py (hunk 5 above)
Comment (hosekadam) [reply convert2rhel/actions/post_conversion/modified_rpm_files_diff.py:82]: Yeah, makes sense, I'll fix that in different PR, since this is already merged

Comment (hosekadam) [reply convert2rhel/actions/post_conversion/modified_rpm_files_diff.py:82]: https://github.com/oamg/convert2rhel/pull/1333
